### ISSUE TITLE
Download with API key

### ISSUE THREAD

**Issue by rwong2888** [XcodesOrg/xcodes#293]
Is there a way to download via API key .p8 file instead of user/pass and 2fa code?

**Comment by adelindanaila** [comment]
+1 

will be great to have option to log in with app store connect api key

**Comment by NajiNasimi-AIYou** [comment]
Any update on this?

**Comment by szagi** [comment]
Hi, unfortunately, in our CI pipeline we use API key instead of FASTLANE_SESSION and at the moment it is impossible to change it, so is it planned to support this authentication method?

**Comment by MattKiazyk** [comment]
Sorry, if there was a way that we could use the ASC api key or a app specific password we would immediately support it. To my knowledge, and last time I tried to look into hacking it, it isn't.

We don't want to have to manually login and 2fa, but Apple doesn't make it easy. At one point, Apple had it open to allow downloads without even logging in (like they currently do with platforms) but they stopped allowing it after a few weeks.

**Comment by StephenHodgson** [comment]
api key support would be greatly appreciated

**Comment by ykhandelwal913** [comment]
api key support would be greatly appreciated

**Comment by ykhandelwal913** [comment]
@MattKiazyk  is this something planned for xcodes install support?

**Comment by olejnjak** [comment]
AFAIK this is not something that is available and is not implemented in Xcodes, it is just not possible with current API

**Comment by StephenHodgson** [comment]
That is correct, the auth is though the website which doesn't share any relation to app store connect api key unfortunately.
